src/vmm/src/devices/virtio/block/virtio/device.rs

```diff
@@ -155,20 +153,17 @@ impl DiskProperties {
         }
         default_id
     }
+}
 
-    /// Provides vec containing the virtio block configuration space
-    /// buffer. The config space is populated with the disk size based
-    /// on the backing file size.
-    pub fn virtio_block_config_space(&self) -> Vec<u8> {
-        // The config space is little endian.
-        let mut config = Vec::with_capacity(BLOCK_CONFIG_SPACE_SIZE);
-        for i in 0..BLOCK_CONFIG_SPACE_SIZE {
-            config.push(((self.nsectors >> (8 * i)) & 0xff) as u8);
-        }
-        config
-    }
+#[derive(Debug, Default, Clone, Copy, Eq, PartialEq)]
+#[repr(C)]
+pub struct ConfigSpace {
+    pub capacity: u64,
 }
 
+// SAFETY: `ConfigSpace` contains only PODs in `repr(C)` or `repr(transparent)`, without padding.
+unsafe impl ByteValued for ConfigSpace {}
```

Comment:
nit: could use `zerocopy` instead of `vm_memory` here (there's long-term plans to remove `ByteValued` from vm_memory and migrate it to zerocopy too)

Reply:
let's do this in one go in another PR

Reply:
yeah, can definitely defer it to later :)